kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go

```diff
@@ -44,8 +44,7 @@ func (source *KafkaSource) ConvertTo(ctx context.Context, obj apis.Convertible)
 			Topics:        source.Spec.Topics,
 			ConsumerGroup: source.Spec.ConsumerGroup,
 		}
-		sink.Status.Status = source.Status.Status
-		source.Status.Status.ConvertTo(ctx, &sink.Status.Status)
+		source.Status.Status.DeepCopyInto(&sink.Status.Status)
```

Comment:
Instead of `.ConvertTo`, I used `DeepCopyInto` since `ConvertTo` was removing the conditions other than `Ready`.
I see status is handled like that in other sources, for example in here: https://github.com/knative/eventing/blob/master/pkg/apis/sources/v1alpha2/apiserver_conversion.go#L85